datastore/gcloud/aio/datastore/datastore_operation.py

```diff
@@ -0,0 +1,28 @@
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+
+class DatastoreOperation:
+    def __init__(self, name: str, done: bool,
+                 metadata: Optional[Dict[str, Any]] = None,
+                 error=None, response: Optional[Dict[str, Any]] = None):
+        self.name = name
+        self.metadata = metadata
+        self.done = done
+        self.error = error
+        self.response = response
+
+    @classmethod
+    def from_repr(cls, data: Dict[str, Any]) -> 'DatastoreOperation':
+        return cls(data.get('name'), data.get('done'), data.get('metadata'),
```

Comment:
Let's do direct indexing instead of `get`s for any of the non-optional fields, to make errors more obvious upfront. Looks like that's `name` and `done`, judging by your `__init__` type hints.